### PR TITLE
Introduce test-failure-policy

### PR DESCRIPTION
**Pull request by marusak**
This is simplified and cleaned up version of tests-policy.
We cannot directly replace this script as specific version of run-tests
depends on it.

Testing for now.

### CODE REVIEW

**Comment by martinpitt** [comment]
From the peanut gallery: This does not really "validate" a test, honestly I don't like the name. How about "test-failure-policy"? (Deliberately not calling it "tests", as this handles a single result at a time). You could even call it "test-policy" (no no, that wouldn't be confusing *at all* :grin: )

**Comment by marusak** [comment]
heh right, naming was the the part where I had no good idea :)
I can call it `test-failure-policy`

**Comment by marusak** [comment]
renamed + made commit message a bit more verbose.

File: task/test-failure-policy [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+Journal extracted to TestMultiMachine-testFrameNavigation-fedora-i386-127.0.0.2-2502-FAIL.log
+"""
+
+
+class TestPolicy(unittest.TestCase):
+    def testKnownIssue(self):
+        script = os.path.join(BOTS_DIR, "test-failure-policy")
+        cmd = [script, "--offline", "example"]
+        proc = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
+        (output, unused) = proc.communicate(PCP_CRASH)
```
Comment (martinpitt) [inline task/test-failure-policy:59]: Either s/unused/_/ or output = ...[0], or use `subprocess.run()`, which is now available everywhere and a bit more comfortable.

File: task/test-failure-policy [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        self.assertEqual(output, "Known issue #9876\n")
```
Comment (martinpitt) [inline task/test-failure-policy:60]: Please also assert the returncode 77 here.

File: task/test-failure-policy [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+        proc = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
+        (output, unused) = proc.communicate("""
+> log: phase coils are misaligned by 0.34 micron
+Traceback (most recent call last):
+  File "test/verify/check-warp-drive", line 34, in testCoils
+     self.assertTrue(all_in_order)
+not ok 1 test/verify/check-warp-drive TestDrive.testCoils
+""")
+        self.assertEqual(output, "Known issue #123\n")
```
Comment (martinpitt) [inline task/test-failure-policy:85]: Please also add test cases for "do nothing" (one for a passing, one for a failing test), with empty output and exit code 0; and a "retry" case with code 1.

File: test-failure-policy [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+from lib.constants import BOTS_DIR
+from lib.testmap import get_test_image
+
+
+def main():
+    script = os.path.basename(__file__)
+    parser = argparse.ArgumentParser(description='Check a traceback for a known issue')
+    parser.add_argument('-o', "--offline", action='store_true',
+                        help="Work offline, don't fetch new data or contact servers")
+    parser.add_argument('image', help="The image to check against")
```
Comment (martinpitt) [inline test-failure-policy:41]: Was it a conscious decision to replace context with image? I wonder if that makes the caller's job harder -- right now you can just use `$TEST_OS`, and tests-policy would strip off the scenario.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks!

File: test-failure-policy (hunk 4 above)
Comment (marusak) [reply test-failure-policy:41]: It was. 

- It was really only interested in image
- Tests only pass image name, never scenario
- The parser in `tests-policy` is not correct (it works because it is called only with image name)
- Tests-policy assumes that default contex looks like `verify/<OS>` which hasn't been true for a while now

So I thought that rather than doing the whole parsing of context etc. just so we can call it manually a bit differently felt a bit pointless so I simplified it as much as possible. You can just do `./test-failure-policy fedora-33` instead of `./test-failure-policy fedora-33/firefox`

File: task/test-failure-policy (hunk 3 above)
Comment (marusak) [reply task/test-failure-policy:126]: done

File: test-failure-policy (hunk 4 above)
Comment (martinpitt) [reply test-failure-policy:41]: If the tests only pass the image name, and that's not much harder than passing `$TEST_OS`, then agreed. Thanks!

**Review by martinpitt — APPROVED** [review]
Thanks! As-is this is safe to land as it only introduces a new script.
